modules/disk-uploader/pkg/certificate/certificate_test.go

```diff
@@ -0,0 +1,117 @@
+package certificate_test
+
+import (
+	"context"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	v1beta1 "kubevirt.io/api/export/v1beta1"
+	kubevirtfake "kubevirt.io/client-go/generated/kubevirt/clientset/versioned/fake"
+	"kubevirt.io/client-go/kubecli"
+
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/kubevirt/kubevirt-tekton-tasks/modules/disk-uploader/pkg/certificate"
+)
+
+var _ = Describe("Certificate", func() {
+	var (
+		vmExportClient *kubevirtfake.Clientset
+		virtClient     kubecli.KubevirtClient
+	)
+
+	const (
+		namespace = "test-namespace"
+		name      = "test-vmexport"
+	)
+
+	BeforeEach(func() {
+		ctrl := gomock.NewController(GinkgoT())
+		vmExportClient = kubevirtfake.NewSimpleClientset()
+
+		kubecli.GetKubevirtClientFromClientConfig = kubecli.GetMockKubevirtClientFromClientConfig
+		kubecli.MockKubevirtClientInstance = kubecli.NewMockKubevirtClient(ctrl)
+		kubecli.MockKubevirtClientInstance.EXPECT().VirtualMachineExport(namespace).Return(vmExportClient.ExportV1beta1().VirtualMachineExports(namespace)).AnyTimes()
+
+		virtClient, _ = kubecli.GetKubevirtClientFromClientConfig(nil)
+	})
+
+	Describe("GetCertificateFromVirtualMachineExport", func() {
+		It("should return the certificate content", func() {
+			_, err := vmExportClient.ExportV1beta1().VirtualMachineExports(namespace).Create(context.Background(),
+				&v1beta1.VirtualMachineExport{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      name,
+						Namespace: namespace,
+					},
+					Status: &v1beta1.VirtualMachineExportStatus{
+						Links: &v1beta1.VirtualMachineExportLinks{
+							Internal: &v1beta1.VirtualMachineExportLink{
+								Cert: "test-cert-content",
+							},
+						},
+					},
+				},
+				metav1.CreateOptions{},
+			)
+			Expect(err).NotTo(HaveOccurred())
+
+			cert, err := certificate.GetCertificateFromVirtualMachineExport(virtClient, namespace, name)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(cert).To(Equal("test-cert-content"))
+		})
+
+		It("should return not found error", func() {
+			_, err := certificate.GetCertificateFromVirtualMachineExport(virtClient, namespace, "test")
+			Expect(err).To(HaveOccurred())
+			Expect(errors.IsNotFound(err)).To(BeTrue())
```

Comment:
You can use `MatchError(errors.IsNotFound, "errors.IsNotFound")` here.

Reply:
I like the usage of MatchError(). Why is better than using `Expect(errors.IsNotFound(err)).To(BeTrue())` - That is widely also used in kubevirt/kubevirt?

Reply:
Also in MatchError() you need to mention not found twice: `errors.IsNotFound, "errors.IsNotFound"`

Reply:
It's also good to have one line instead of two lines of Expect().